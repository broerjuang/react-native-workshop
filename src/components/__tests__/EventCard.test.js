// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import {EventCard, renderIconType} from '../index';
import type {Event} from '../../features/events/types/Event';

describe('EventCard', () => {
  it('should render EventCard with comment correctly', () => {
    let event = {
      type: 'IssueCommentEvent',
      actor: {
        login: 'zzzcielo',
        avatar_url:
          'http://www.grosse.is-a-geek.com/robopics/roborovski01_1024.jpg',
      },
      repo: {
        id: '123',
        name: 'astridtamara/bootcamp',
      },
      payload: {
        action: 'comment',
        issue: {
          title: 'Parallax Header bug',
        },
        comment: {},
      },
      created_at: '1d',
    };
    let component = renderer.create(
      <EventCard event={event} navigateScreen={() => {}} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    event.payload.comment['body'] = 'test';
    component = renderer.create(
      <EventCard event={event} navigateScreen={() => {}} />,
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render EventCard of type IssuesEvent correctly', () => {
    let event = {
      type: 'IssuesEvent',
      actor: {
        login: 'zzzcielo',
        avatar_url:
          'http://www.grosse.is-a-geek.com/robopics/roborovski01_1024.jpg',
      },
      repo: {
        id: '123',
        name: 'astridtamara/bootcamp',
      },
      payload: {
        action: 'closed',
        issue: {
          title: 'Bug Fix',
        },
      },
      created_at: '1d',
    };
    let component = renderer.create(
      <EventCard event={event} navigateScreen={() => {}} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    event.payload.action = 'reopened';
    component = renderer.create(
      <EventCard event={event} navigateScreen={() => {}} />,
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    event.payload.action = 'opened';
    component = renderer.create(
      <EventCard event={event} navigateScreen={() => {}} />,
    );
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Icon according to Type correctly', () => {
    let event: Event = {
      type: 'PushEvent',
      actor: {
        login: 'astridtamara',
        avatar_url: '',
      },
      repo: {
        id: '9',
        name: 'astridtamara/bootcamp',
      },
      payload: {
        ref: 'test',
      },
      created_at: '1d',
    };
    let component = renderer.create(<renderIconType event={event} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
