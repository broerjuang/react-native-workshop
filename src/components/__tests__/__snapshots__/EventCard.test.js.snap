// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render EventCard corectly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#d6d4d4",
      "borderWidth": 0.5,
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Array [],
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "width": "15%",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "http://www.grosse.is-a-geek.com/robopics/roborovski01_1024.jpg",
            }
          }
          style={
            Object {
              "borderRadius": 16,
              "height": 32,
              "width": 32,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "flexWrap": "wrap",
            "justifyContent": "center",
            "padding": 10,
            "width": "70%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={[Function]}
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            zzzcielo
             
          </Text>
          commented on issue
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={[Function]}
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
             
            Parallax Header Bug
             
          </Text>
          at
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            onPress={[Function]}
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
             
            astridtamara/bootcamp
             
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "grey",
              }
            }
          >
            1d
          </Text>
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "width": "15%",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    numberOfLines={3}
    style={
      Object {
        "color": "grey",
        "fontWeight": "bold",
        "padding": 12,
        "paddingTop": 0,
      }
    }
  >
    [Test Long Comment] The bug is bugging me a lot. Good job fixing it in such short time.[Test Long Comment] The bug is bugging me a lot. Good job fixing it in such short time.[Test Long Comment] The bug is bugging me a lot. Good job fixing it in such short time.[Test Long Comment] The bug is bugging me a lot. Good job fixing it in such short time. 
  </Text>
</View>
`;
